cluster/examples/kubernetes/ceph/create-external-cluster-resources.py

```diff
@@ -0,0 +1,633 @@
+'''
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+'''
+
+import sys
+import json
+import argparse
+import unittest
+import re
+from os import linesep as LINESEP
+
+# backward compatibility with 2.x
+try:
+    ModuleNotFoundError
+except:
+    ModuleNotFoundError = ImportError
+
+try:
+    import rados
+except ModuleNotFoundError as noModErr:
+    print("Error: %s\nExiting the script..." % noModErr)
+    sys.exit(1)
+
+try:
+    # for 2.7.x
+    from StringIO import StringIO
+except ModuleNotFoundError:
+    # for 3.x
+    from io import StringIO
+
+
+class ExecutionFailureException(Exception):
+    pass
+
+
+class RadosJSON:
+    EXTERNAL_USER_NAME = "client.healthchecker"
+    EMPTY_OUTPUT_LIST = "Empty output list"
+
+    @classmethod
+    def gen_arg_parser(cls, args_to_parse=None):
+        argP = argparse.ArgumentParser()
+        argP.add_argument("--verbose", "-v",
+                          action='store_true', default=False)
+        argP.add_argument("--ceph-conf", "-c",
+                          help="Provide a ceph conf file.", type=str)
+        argP.add_argument("--run-as-user", "-u",
+                          help="Provides a user name to check the cluster's health status",
+                          default=cls.EXTERNAL_USER_NAME, type=str)
+        argP.add_argument("--format", "-t", choices=["json", "bash"],
+                          default='json', help="Provides the output format (json | bash)")
+        argP.add_argument("--cluster-name", default="openshift-storage",
+                          help="Ceph cluster name")
+        argP.add_argument("--output", "-o", default="",
+                          help="Output will be stored into the provided file")
+        argP.add_argument("--cephfs-filesystem-name", default="",
+                          help="Provides the name of the Ceph filesystem")
+        argP.add_argument("--cephfs-data-pool-name", default="",
+                          help="Provides the name of the cephfs data pool")
+        argP.add_argument("--rbd-data-pool-name", default="", required=True,
```

Comment:
We need to make sure the specified pool exists.

Reply:
Yup, it is checked through following snippet of code,

https://github.com/rook/rook/blob/2db9b21ddd8cbac744bd261142d3ffe7e772f1c5/cluster/examples/kubernetes/ceph/create-external-cluster-resources.py#L255

Reply:
If I run the script with `--rbd-data-pool-name <random name>`, it does not through an error, it should since the pool does not exist in the cluster.

Reply:
Yup, got it @leseb , I was mostly referring to the `cephfs-data-pool-name`...
Added the check...

Reply:
https://github.com/rook/rook/blob/23d219114758097691f0c86f7123161f03182925/cluster/examples/kubernetes/ceph/create-external-cluster-resources.py#L306